src/MongoDB.Driver/PipelineStageDefinitionBuilder.cs

```diff
@@ -489,8 +489,8 @@ public static class PipelineStageDefinitionBuilder
             Ensure.IsNotNull(connectToField, nameof(connectToField));
             Ensure.IsNotNull(startWith, nameof(startWith));
             Ensure.IsNotNull(@as, nameof(@as));
-            Ensure.That(IsTConnectToOrEnumerableTConnectTo<TConnectFrom, TConnectTo>(), "TConnectFrom must be either TConnectTo or a type that implements IEnumerable<TConnectTo>.", nameof(TConnectFrom));
-            Ensure.That(IsTConnectToOrEnumerableTConnectTo<TStartWith, TConnectTo>(), "TStartWith must be either TConnectTo or a type that implements IEnumerable<TConnectTo>.", nameof(TStartWith));
+            Ensure.That(IsOneToOneOrOneToManyRelation<TConnectFrom, TConnectTo>(), "TConnectFrom must be TConnectTo or TConnectFrom must be a type that implements IEnumerable<TConnectTo> or TConnectTo must be a type that implements IEnumerable<TConnectFrom>.", nameof(TConnectFrom));
+            Ensure.That(IsOneToOneOrOneToManyRelation<TStartWith, TConnectTo>(), "TStartWith must be TConnectTo or TStartWith must be a type that implements IEnumerable<TConnectTo> or TConnectTo must be a type that implements IEnumerable<TStartWith>.", nameof(TStartWith));
```

Comment:
Please pay extra attention to these error messages. They seem to be too long. I don't have ideas how to make them smaller if it is needed.